### PR TITLE
Passive compiler, finishing up

### PR DESCRIPTION
**Pull request by jakeffbulmer**
Tidying up on passive compiler PR

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -79,6 +79,52 @@
   prog_merged = prog.compile(compiler="gaussian_merge")
   ```
 
+* A new operation, ``PassiveChannel`` has been added. It allows for arbitrary linear/passive transformations (i.e. any operation which is linear in creation operators). Currently only supported by the ``gaussian`` backend. [(#600)](https://github.com/XanaduAI/strawberryfields/pull/600)
```
Comment (josh146) [inline .github/CHANGELOG.md:82]:
```suggestion
* A new operation, ``PassiveChannel`` has been added. It allows for arbitrary linear/passive transformations
  (i.e., any operation which is linear in creation operators). Currently only supported by the ``gaussian``
  backend. [(#600)](https://github.com/XanaduAI/strawberryfields/pull/600)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -79,6 +79,52 @@
...
+  circuit = sf.Program(M)
+  with circuit.context as q:
+      for i in range(M):
+          ops.Sgate(1) | q[i]
+      ops.PassiveChannel(T) | q
+
+  cov = eng.run(circuit).state.cov()
+  ```
+ 
+* A new compiler, ``passive``, allows for a circuit which only consists of passive elements to be compiled into a single ``PassiveChannel``. [(#600)](https://github.com/XanaduAI/strawberryfields/pull/600)
```
Comment (josh146) [inline .github/CHANGELOG.md:109]:
```suggestion
* A new compiler, ``passive``, allows for a circuit which only consists of passive
  elements to be compiled into a single ``PassiveChannel``.
  [(#600)](https://github.com/XanaduAI/strawberryfields/pull/600)

```

File: strawberryfields/ops.py [hunk 3]
```diff
@@ -1424,16 +1424,18 @@ class PassiveChannel(Channel):
         Acts the following transformation on the state:
 
         .. math::
-            a^{\dagger}_i \to \sum_j T_{ij} a^{\dagger}j
+            a^{\dagger}_i \to \sum_j T_{ij} a^{\dagger}_j
+
```
Comment (josh146) [inline strawberryfields/ops.py:1428]: For some reason, I can't see what changed here?

File: strawberryfields/ops.py [hunk 4]
```diff
@@ -1424,16 +1424,18 @@ class PassiveChannel(Channel):
...
 
     def __init__(self, T):
+        T = np.atleast_2d(T)
         super().__init__([T])
         self.ns = T.shape[0]
 
     def _apply(self, reg, backend, **kwargs):
         p = par_evaluate(self.p)
-        backend.passive(p[0], *reg)
+        backend.passive(p[0], reg)
```
Comment (josh146) [inline strawberryfields/ops.py:1438]: was this a bug?

**Review by josh146 — CHANGES_REQUESTED** [review]
Thanks for getting to this so quickly @jakeffbulmer 💪 Just some small changes, and then it can be merged in

File: strawberryfields/ops.py [hunk 5]
```diff
@@ -1424,16 +1424,18 @@ class PassiveChannel(Channel):
         Acts the following transformation on the state:
 
         .. math::
-            a^{\dagger}_i \to \sum_j T_{ij} a^{\dagger}j
+            a^{\dagger}_i \to \sum_j T_{ij} a^{\dagger}_j
+
     """
 
     def __init__(self, T):
+        T = np.atleast_2d(T)
```
[the remark this answers is not in the document]
Comment (jakeffbulmer) [reply strawberryfields/ops.py:1432]: pretty handy right!? Also np.atleast_1d :D

File: strawberryfields/ops.py (hunk 3 above)
Comment (jakeffbulmer) [reply strawberryfields/ops.py:1428]: i think all i changed was just the underscore on the final a^{\dagger}

File: strawberryfields/ops.py (hunk 4 above)
Comment (jakeffbulmer) [reply strawberryfields/ops.py:1438]: no, but i realised that i didn't need to unpack, so i removed it here and in the backend function too

**Review by josh146 — APPROVED** [review]
💯

**Pull request merged**